File: DataStructurePython/stack.py
class Stack:
    def __init__(self):
        self.array = []

    def is_empty(self):
        return len(self.array) == 0

    def push(self, item):
        self.array.append(item)

    def pop(self):
        if self.is_empty():
            return
        return self.array.pop()

    def peek(self):
        if self.is_empty():
            return
        return self.array[-1]

File: DataStructureSwift/Sources/DataStructureSwift/Stack.swift
//
//  Stack.swift
//  
//
//  Created by Yubo Qin on 11/29/22.
//

import Foundation

class Stack<T> {

    private var array: [T]

    init() {
        self.array = []
    }

    var isEmpty: Bool {
        array.isEmpty
    }

    func push(_ item: T) {
        array.append(item)
    }

    @discardableResult
    func pop() -> T? {
        return array.popLast()
    }

    func peek() -> T? {
        return array.last
    }

}

extension Stack: CustomStringConvertible {

    var description: String {
        return "Stack: \(array)"
    }

}
